lib/ansible/modules/cloud/amazon/rds_cluster.py

```diff
@@ -0,0 +1,843 @@
+                response = client.delete_db_cluster(**params)
+                changed = True
+            except botocore.exceptions.ClientError as e:
+                e = get_exception()
+                module.fail_json(msg=str(e))
+
+            if module.params.get('wait'):
+                #wait for status deleting, then wait for status deleted
+                status_achieved, response = wait_for_status(client, module, 'deleting')
+                status_achieved, response = wait_for_status(client, module, 'deleted')
+                if not status_achieved:
+                    module.fail_json(msg='Error deleting RDS cluster  - please check the AWS console')
+
+    result = response
+    return changed, result
+
+
+def validate_parameters(required_vars, valid_vars, module):
+    params = {}
+    command = module.params.get('command')
+
+    # convert snek case args to camel case params required for API
+    camel_params = {
+        'availability_zones': 'AvailabilityZones',
+        'backup_retention_period': 'BackupRetentionPeriod',
+        'character_set_name': 'CharacterSetName',
+        'database_name': 'DatabaseName',
+        'db_cluster_identifier': 'DBClusterIdentifier',
+        'db_cluster_parameter_group_name': 'DBClusterParameterGroupName',
+        'db_cluster_snapshot_identifer': 'DBClusterSnapshotIdentifier',
+        'vpc_security_group_ids': 'VpcSecurityGroupIds',
+        'new_db_cluster_identifier': 'NewDBClusterIdentifier',
+        'db_subnet_group_name': 'DBSubnetGroupName',
+        'engine': 'Engine',
+        'engine_version': 'EngineVersion',
+        'port': 'Port',
+        'master_username': 'MasterUsername',
+        'master_user_password': 'MasterUserPassword',
+        'option_group_name': 'OptionGroupName',
+        'preferred_maintenance_window': 'PreferredMaintenanceWindow',      
+        'preferred_backup_window': 'PreferredBackupWindow',
+        'replication_source_identifier': 'ReplicationSourceIdentifier',
+        'tags': 'Tags',
+        'storage_encrypted': 'StorageEncrypted',
+        'kms_key_id': 'KmsKeyId',
+        'skip_final_snapshot': 'SkipFinalSnapshot',
+        'final_db_snapshot_identifier': 'FinalDBSnapshotIdentifier',
+        'apply_immediately': 'ApplyImmediately',
+        'snapshot_identifer': 'SnapshotIdentifier'}
+
+    for (k, v) in camel_params.iteritems():
+        if module.params.get(k) and k not in required_vars:
+            if k in valid_vars: 
+                params[v] = module.params[k]
+            else:
+                module.fail_json(msg="Parameter %s is not valid for %s command" % (k, command))
+        if k in required_vars:
+            if module.params.get(k):
+                params[v] = module.params[k]
+            else:
+                module.fail_json(msg="Parameter %s required for %s command" % (k, command))
+
+    tag_array = []
+    if module.params.get('tags'):
+        for tag, value in module.params.get('tags').iteritems():
```

Comment:
Should be `for tag, value in module.params.get('tags').items()
` as `.iteritems` isn't universal.